src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -153,16 +153,6 @@ public static int Main(string[] args)
                 log.Error("Exception in RemoteTestAgent", ex);
             }
 
-            try
-            {
-                // Unregister the channel
-                testAgencyServer.Stop();
```

Comment:
Maybe the confusion is that the same `ServerBase` type is instantiated twice- one on the server side, one on the client side. The fix definitely has to be there inside the implementation of `Stop` so that when `Stop` executes on the remote side, it doesn't close the socket too soon. When `Stop` is called on the test process side, none of the code in `Stop` actually runs in the test process.